libs/extensions/src/NanoDimuonVertex.cpp

```diff
@@ -51,9 +52,19 @@ pair<shared_ptr<PhysicsObject>,shared_ptr<PhysicsObject>> NanoDimuonVertex::GetM
   return make_pair(muon1,muon2);
 }
 
+float NanoDimuonVertex::GetDimuonChargeProduct(const shared_ptr<Event> event) {
+  auto muons = GetMuons(event);
+  return float(muons.first->GetAsFloat("charge")) * float(muons.second->GetAsFloat("charge"));
+}
+
 bool NanoDimuonVertex::PassesChi2Cut() { return (float)Get("chi2") < muonVertexCuts["maxChi2"]; }
 
-bool NanoDimuonVertex::PassesDeltaRCut() { return (float)Get("dR") < muonVertexCuts["maxDeltaR"]; }
+bool NanoDimuonVertex::PassesMaxDeltaRCut() { return (float)Get("dR") < muonVertexCuts["maxDeltaR"]; }
+bool NanoDimuonVertex::PassesMinDeltaRCut(const shared_ptr<Event> event) { 
+  auto muons = GetMuons(event);
+  float dR = asNanoMuon(muons.first)->GetFourVector().DeltaR(asNanoMuon(muons.second)->GetFourVector());
+  return dR > muonVertexCuts["minDeltaR"];
+}
```

Comment:
Why for MaxDeltaR we extract dR directly from the vertex, but for MinDelraR we have to get muons from the event and calculate delta R ourselves?

Reply:
Very good question, they should be the same so I will just stick to one of them